src/Microsoft.Health.Fhir.Core/Features/Definition/SearchParameterDefinitionBuilder.cs

```diff
@@ -115,31 +116,34 @@ internal void Build()
             _initialized = true;
         }
 
-        private static bool ShouldExcludeEntry(string resourceType, string searchParameterName)
+        private bool ShouldExcludeEntry(string resourceType, string searchParameterName)
         {
-            return (resourceType == ResourceType.DomainResource.ToString() && searchParameterName == "_text") ||
-                   (resourceType == ResourceType.Resource.ToString() && searchParameterName == "_content") ||
-                   (resourceType == ResourceType.Resource.ToString() && searchParameterName == "_query")
-#if Stu3
-                || (resourceType == ResourceType.DataElement.ToString() && (searchParameterName == "objectClass" || searchParameterName == "objectClassProperty"))
-#endif
-                ;
+            return (resourceType == KnownResourceTypes.DomainResource && searchParameterName == "_text") ||
+                   (resourceType == KnownResourceTypes.Resource && searchParameterName == "_content") ||
+                   (resourceType == KnownResourceTypes.Resource && searchParameterName == "_query") ||
+                   ShouldExcludeEntryStu3(resourceType, searchParameterName);
+        }
+
+        private bool ShouldExcludeEntryStu3(string resourceType, string searchParameterName)
+        {
+            return _modelInfoProvider.Version == FhirSpecification.Stu3 &&
+                   (resourceType == "DataElement" &&
```

Comment:
Should DataElement also be a KnownResourceType? 
Nit: The parentheses on this line aren't needed.